mitodl/mitxpro
# Data consent checkbox

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #216 

#### What's this PR do?
Adds a data consent checkbox and UI to view the data consent agreement. The checkbox should not be visible if the data consent agreement coupon has not been applied. If the checkbox exists but is false, a validation error should tell the user to check it.

#### How should this be manually tested?
 - Create a coupon which is attached to a company using the form at `/ecommerce/admin/coupons/`. Make it apply to some valid purchasable target.
 - Check out with the same product. You should not see a data consent agreement checkbox.
 - Apply the coupon at checkout. You should see the data consent agreement checkbox soon after.
 - Submit the page without checking the box. You should see a validation error.
 - Click the checkbox and submit the page. You should be redirected to Cybersource (or back to `/dashboard/` for a 100% off coupon). 
 - Verify that the `DataConsentUser` object for your user and for the coupon you used has a date for `consent_date` which is approx the time you submitted the checkout page.

#### Screenshots

Checkbox with validation
![Screenshot from 2019-06-12 16-11-23](https://user-images.githubusercontent.com/863262/59382907-d39efc00-8d2c-11e9-956c-b702bb1fce30.png)

Data consent agreement text
![Screenshot from 2019-06-12 16-11-39](https://user-images.githubusercontent.com/863262/59382912-d863b000-8d2c-11e9-99a7-061a51fc66da.png)

Mobile width agreement text
![Screenshot from 2019-06-12 16-13-35](https://user-images.githubusercontent.com/863262/59383013-152fa700-8d2d-11e9-83f1-f30df92b4c3a.png)
